### PR TITLE
🧹 Cleanup a few `TODO`s

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area dev-productivity
/kind cleanup technical-debt

**What this PR does / why we need it**:
This PR cleans up the following `TODO`s from the code:

- Drop `allow-{to,from}-shoot-apiserver` `NetworkPolicy`s (deprecated with https://github.com/gardener/gardener/pull/7515, released since `v1.65.0`)
- Drop `allow-{from-}prometheus` `NetworkPolicy`s (deprecated with https://github.com/gardener/gardener/pull/7484, released since `v1.65.0`)
- Drop finalizer removal migration code (introduced with https://github.com/gardener/gardener/pull/7636, released since `v1.67.0`)
- Drop labeling of ETCD encryption config secrets (introduced with https://github.com/gardener/gardener/pull/7244, released since `v1.63.0`)
- Drop reseting worker state when it contained `null`-ed fields (introduced with https://github.com/gardener/gardener/pull/7637, released since `v1.67.0`)
- No longer label Terraformer with `to-seed-apiserver` (replaced with https://github.com/gardener/gardener/pull/7389, released since `v1.64.0`)
- Drop cleanup of `ManagedResource`s for `gardener-seed-admission-controller` (introduced with https://github.com/gardener/gardener/pull/7053, released since `v1.61.0`)
- Drop cleanup of `allow-{to-loki,loki,from-aggregate-prometheus}` `NetworkPolicy`s (introduced with https://github.com/gardener/gardener/pull/7636, released since `v1.67.0`)
- Drop cleanup of `ManagedResource` for shoot addons (introduced with https://github.com/gardener/gardener/pull/7574, released since `v1.66.0`)
- Drop cleanup of `allow-{to-loki,from-prometheus-to-loki-telegraf}` `NetworkPolicy`s (introduced with https://github.com/gardener/gardener/pull/7515, released since `v1.65.0`)
- Drop cleanup of `generic-token-kubeconfig` secret (introduced with https://github.com/gardener/gardener/pull/6891, released since `v1.59.0`)
- Drop cleanup of `allow-etcd` `NetworkPolicy` (introduced with https://github.com/gardener/gardener/pull/7515, released since `v1.65.0`)
- Drop cleanup of `allow-kube-apiserver-to-{gardener-resource-manager,vpa-admission-controller}` `NetworkPolicy`s (introduced with https://github.com/gardener/gardener/pull/7515, released since `v1.65.0`)
- Drop cleanup of `allow-to-vpn-seed-server` `NetworkPolicy` (introduced with https://github.com/gardener/gardener/pull/7515, released since `v1.65.0`)
- Drop legacy `apiserver-proxy` Helm chart (replaced with https://github.com/gardener/gardener/pull/7547, released since `v1.66.0`)
- Drop `NetworkPolicy` migration code on `gardenlet` start-up (last modified with https://github.com/gardener/gardener/pull/7570, released since `v1.66.0`)
- Enforce non-internal API versions for all provider configs (introduced with https://github.com/gardener/gardener/pull/4927, released since `v1.38.0`)
- Drop `failureToleranceType != nil` condition in HA config webhook (changed with https://github.com/gardener/gardener/pull/6997, released since `v1.63.0`)
- Check for `status != True` in `Etcd` health check (introduced with https://github.com/gardener/gardener/pull/6552, released since `etcd-druid@v0.15.0`)
- Drop fallback to `LastUpdateTime` of extension conditions (changed with https://github.com/gardener/gardener/pull/6626, released since `v1.58.0`)
- Drop cleanup of legacy zone enforcement label (introduced with https://github.com/gardener/gardener/pull/6934, released since `v1.59.0`)
- Drop migration worker settings for `etcd-druid` (introduced with https://github.com/gardener/gardener/pull/5037, released since `v1.37.0`)
- Add missing ignore annotations for istio resources (those were forgotten in https://github.com/gardener/gardener/pull/7397 and hinder cleanup right now)
- ~~Drop migration code for `istio` `ManagedResource` split (introduced with https://github.com/gardener/gardener/pull/7397, released since `v1.65.0`)~~
- Drop `nodes/stats` permission for `metrics-server` (no longer needed with https://github.com/gardener/gardener/pull/6987, released since `v1.61.0`)
- Drop `networkpolicies` component (deprecated with https://github.com/gardener/gardener/pull/7636, released since `v1.67.0`)
- Drop cleanup of orphaned `DNSRecord` secrets (introduced with https://github.com/gardener/gardener/pull/4926, released since `v1.38.0`)
- Switch ETCD from `to-seed-apiserver` to `to-runtime-apiserver` label (introduced with https://github.com/gardener/gardener/pull/7434, released since `etcd-druid@v0.17.0`)
- Drop custom `allow-etcd-peer` `NetworkPolicy` (introduced with https://github.com/gardener/gardener/pull/7515, released since `etcd-druid@v0.17.0`)
- Drop `identity` value from Helm chart values (replaced with https://github.com/gardener/gardener/pull/7624, released since `v1.67.0`)
- Remove TODO for dropping zone pinning migration code
- Drop empty `NetworkConfig` for calico for local shoots
- Remove actionless TODOs
- Use `scale` subresource for scaling

Part of https://github.com/gardener/gardener/issues/7352

**Release note**:

```breaking developer
Extensions vendoring this `gardener/gardener` version need to provide RBAC privileges for `PATCH apps/depoyments/scale`.
```
```breaking operator
The deprecated `allow-{to,from}-shoot-apiserver` `NetworkPolicy`s have been dropped. Ensure that all registered extensions have been adapted.
```
```breaking operator
The deprecated `identity` value is no longer passed when `ControllerInstallation` Helm charts are deployed.
```
```breaking operator
The `lastUpdateTime` of extension conditions is no longer considered. Ensure that all registered extensions populate the `lastHeartbeatTime` field instead.
```
```breaking user
Using internal API versions in `providerConfig` fields is no longer permitted (deprecated since more than `2y`). Ensure that you always use a versioned API.
```

### CODE REVIEW

**Review by shafeeqes — COMMENTED** [review]
Thanks a lot for the effort.
/lgtm

File: cmd/gardenlet/app/app.go [hunk 1]
```diff
@@ -452,173 +437,6 @@ func (g *garden) registerSeed(ctx context.Context, gardenClient client.Client) e
...
-			if withGardenNamespaceSelector {
-				selectors = append(selectors, metav1.LabelSelector{MatchLabels: map[string]string{corev1.LabelMetadataName: v1beta1constants.GardenNamespace}})
-			}
-
-			selectors = append(selectors,
-				metav1.LabelSelector{MatchLabels: map[string]string{v1beta1constants.GardenRole: v1beta1constants.GardenRoleIstioIngress}},
-				metav1.LabelSelector{MatchExpressions: []metav1.LabelSelectorRequirement{{Key: v1beta1constants.LabelExposureClassHandlerName, Operator: metav1.LabelSelectorOpExists}}},
-			)
-
-			if withGardenNamespaceSelector && features.DefaultFeatureGate.Enabled(features.FullNetworkPoliciesInRuntimeCluster) {
```
Comment (timuthy) [inline cmd/gardenlet/app/app.go]: I'm not too sure about this but don't we need this code until the `features.FullNetworkPoliciesInRuntimeCluster` feature gate is locked to `true`?

File: plugin/pkg/shoot/validator/admission.go [hunk 2]
```diff
@@ -283,10 +283,7 @@ func (v *ValidateShoot) Admit(ctx context.Context, a admission.Attributes, o adm
 	validationContext.addMetadataAnnotations(a)
 
 	var allErrs field.ErrorList
-	// TODO (voelzmo): remove this 'if' statement once we gave owners of existing Shoots a nice grace period to move away from 'internal' apiVersion
-	if a.GetOperation() == admission.Create {
-		allErrs = append(allErrs, validationContext.validateAPIVersionForRawExtensions()...)
-	}
+	allErrs = append(allErrs, validationContext.validateAPIVersionForRawExtensions()...)
```
Comment (timuthy) [inline plugin/pkg/shoot/validator/admission.go:286]: Should we have a breaking release note for this?

**Review by timuthy — COMMENTED** [review]
Nice cleanup 🙂 Only two concerns and esp. one I'm not sure about.

File: plugin/pkg/shoot/validator/admission.go (hunk 2 above)
Comment (rfranzke) [reply plugin/pkg/shoot/validator/admission.go:286]: 👍🏻 done

File: cmd/gardenlet/app/app.go (hunk 1 above)
Comment (rfranzke) [reply cmd/gardenlet/app/app.go]: Good catch, I reverted this commit. Thanks :)

**Review by timuthy — COMMENTED** [review]
/lgtm

**Review by timuthy — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**